.buildkite/pipeline.yml

```diff
@@ -1,2 +1,46 @@
+agents:
+  image: "docker.elastic.co/ci-agent-images/eck-region/go-builder-buildkite-agent:1.19-02"
+  cpu: "4"
+  memory: "4G"
+
+env:
+  DOCKER_IMAGE: "docker.elastic.co/cloud-ci/k8s-arch/elasticsearch-k8s-metrics-adapter"
+  DOCKER_IMAGE_TAG: "git-${BUILDKITE_COMMIT:0:12}"
+
 steps:
-  - command: echo "🐙"'
\ No newline at end of file
+  - label: ":go: Run unit tests"
+    command: "make test"
+    key: "tests"
+    artifact_paths:
+      - "coverage.out"
+
+  - label: ":sonarqube: Static Code Analysis"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/serverless/shared-analysis-token"
+    agents:
+      image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+    command:
+      - "buildkite-agent artifact download coverage.out ."
+      - "/scan-source-code.sh"
+    soft_fail: true
+    depends_on: "tests"
+
+  # Need go 1.20 in the serverless-docker-builder to enable this 
+  #- label: ":go: Build"
+  #  command: "make all"
+
+  # multiarch dockerx build
+  - group: ":docker: Build Container Images"
```

Comment:
after this you maybe want to immediately push changes out to **dev**, **qa** with a step similar to:
https://github.com/elastic/project-controller/blob/259b8c73e55b81251b7ef07ed110a2799955b74f/.buildkite/pipeline.yml#L99-L105